src/app/forms/components/weights/weights.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormArray } from '@angular/forms';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode } from '@forms/services/dynamic-form.types';
+import { Axle, TechRecordModel } from '@models/vehicle-tech-record.model';
+import { debounceTime, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-weights',
+  templateUrl: './weights.component.html'
+})
+export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() isEditable = false;
+  @Input() template: FormNode | undefined;
+  @Input() data: Partial<TechRecordModel> = {};
```

Comment:
This component will not function without its `data` and `template`. So, you can enforce those in its selector.
```suggestion
@Component({
  selector: 'app-weights[data][template]',
  templateUrl: './weights.component.html'
})
export class WeightsComponent implements OnInit, OnDestroy, OnChanges {
  @Input() isEditable = false;
  @Input() template!: FormNode;
  @Input() data!: Partial<TechRecordModel>;
```

Reply:
Ah this is nice, great tip